src/Stratis.Features.FederatedPeg/Distribution/DistributionManager.cs

```diff
@@ -0,0 +1,138 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using Microsoft.Extensions.Logging;
+using NBitcoin;
+using Stratis.Features.Collateral;
+using Stratis.Features.FederatedPeg.Wallet;
+
+namespace Stratis.Features.FederatedPeg.Distribution
+{
+    /// <summary>
+    /// Constructs the list of recipients for the mainchain reward sharing.
+    /// Runs on the sidechain.
+    /// </summary>
+    public class DistributionManager : IDistributionManager
+    {
+        private const int DefaultEpoch = 240;
+        
+        private readonly Network network;
+        private readonly ChainIndexer chainIndexer;
+        private readonly ILogger logger;
+
+        private int epoch;
+        private int lastDistributionHeight;
+
+        // The reward each miner receives upon distribution is computed as a proportion of the overall accumulated reward since the last distribution.
+        // The proportion is based on how many blocks that miner produced in the period (each miner is identified by their block's coinbase's scriptPubKey).
+        // It is therefore not in any miner's advantage to delay or skip producing their blocks as it will affect their proportion of the produced blocks.
+        // We pay no attention to whether a miner has been kicked since the last distribution or not.
+        // If they produced an accepted block, they get their reward.
+
+        public DistributionManager(Network network, ChainIndexer chainIndexer, ILoggerFactory loggerFactory)
+        {
+            this.network = network;
+            this.chainIndexer = chainIndexer;
+            this.logger = loggerFactory.CreateLogger(this.GetType().FullName);
+
+            this.epoch = this.network.Consensus.MaxReorgLength == 0 ? DefaultEpoch : (int)this.network.Consensus.MaxReorgLength;
+            this.lastDistributionHeight = 0;
```

Comment:
This isnt't used or set anywhere?

Reply:
This is a leftover from the initial implementation, it can be removed